ports/psoc6/mphalport.c

```diff
@@ -63,6 +76,42 @@ int mp_hal_stdin_rx_chr(void) {
 }
 
 
+void mp_hal_pin_od_low(mp_hal_pin_obj_t pin) {
+    gpio_clear_value(pin);
+}
+
+
+void mp_hal_pin_od_high(mp_hal_pin_obj_t pin) {
+    gpio_set_value(pin);
+}
+
+
+int mp_hal_pin_read(mp_hal_pin_obj_t pin) {
+    return gpio_get_value(pin);
+}
+
+
+void mp_hal_pin_open_drain(mp_hal_pin_obj_t pin) {
+    cyhal_gpio_configure(pin, CYHAL_GPIO_DIR_INPUT,  CYHAL_GPIO_DRIVE_OPENDRAINDRIVESLOW);
+
+    // printf("2 SDA is CYHAL_GPIO_DRIVE_OPENDRAINDRIVESLOW ? %ld\n", gpio_get_drive(CYBSP_I2C_SDA));
+    // printf("2 SCL is CYHAL_GPIO_DRIVE_OPENDRAINDRIVESLOW ? %ld\n", gpio_get_drive(CYBSP_I2C_SCL));
+
+    // printf("SDA is CYHAL_GPIO_DRIVE_OPENDRAINDRIVESLOW ? %d\n", (gpio_get_drive(CYBSP_I2C_SDA) & 0xFFFFFFF7) == CYHAL_GPIO_DRIVE_OPENDRAINDRIVESLOW);
+    // printf("SCL is CYHAL_GPIO_DRIVE_OPENDRAINDRIVESLOW ? %d\n", (gpio_get_drive(CYBSP_I2C_SCL) & 0xFFFFFFF7) == CYHAL_GPIO_DRIVE_OPENDRAINDRIVESLOW);
+}
+
+
+uint8_t mp_hal_pin_name(mp_hal_pin_obj_t pin) {
+    return pin;
+}
+
+
+mp_hal_pin_obj_t mp_hal_get_pin_obj(mp_obj_t obj) {
```

Comment:
Passing a wrong pin or invalid pin is not handled in softI2C as it passed through the `mp_hal`. This can be done by putting a check in the `pin_find()` function.

In the hard mode, the pins are passed directly to the `CYHAL I2C` which handles invalid pins.

Reply:
A check for validity of `pin` returned from `pin_find()` is present in the `machine.Pin.make_new()` but since `pin_find()` is now also called from other places except `machine.Pin`, the check is now moved inside 'pin_find()' instead. This solves the problem mentioend above.

Reply:
Agreed as discussed